frame/dapps-staking/src/tests.rs

```diff
@@ -1289,6 +1289,51 @@ fn claim_after_unregister_is_ok() {
         }
     })
 }
+#[test]
+fn claim_only_payout_is_ok() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+
+        let developer = 1;
+        let staker = 2;
+        let contract_id = MockSmartContract::Evm(H160::repeat_byte(0x01));
+
+        let start_era = DappsStaking::current_era();
+        assert_register(developer, &contract_id);
+        let stake_value = 100;
+        assert_bond_and_stake(staker, &contract_id, stake_value);
+
+        advance_to_era(start_era + 1);
+        assert_ok!(DappsStaking::enable_compound_staking(
+            Origin::signed(staker),
+            RewardHandling::OnlyPayout
+        ));
```

Comment:
I suggest you add an `assert_` utility function for this where you assert the event and init/final state.